unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2597,30 +2603,39 @@ protected void updateAllAgentCollidersForVisibilityCheck(bool enableColliders)
         }
 
         public SimObjPhysics[] VisibleSimObjs(string objectId, bool forceVisible = false) {
-            ServerAction action = new ServerAction();
-            action.objectId = objectId;
-            action.forceVisible = forceVisible;
-            return VisibleSimObjs(action);
+            SimObjPhysics sop = getSimObjectFromTypeOrId(objectType: null, objectId: objectId);
+            List<SimObjPhysics> simObjs = new List<SimObjPhysics>();
+            if (sop != null) {
+                simObjs.Add(sop);
+            }
+            if (forceVisible) {
+                return simObjs.ToArray();
+            } else {
+                return GetAllVisibleSimObjPhysics(m_Camera, maxVisibleDistance, simObjs);
+            }
         }
 
 
         public SimObjPhysics[] VisibleSimObjs(ServerAction action) {
             List<SimObjPhysics> simObjs = new List<SimObjPhysics>();
+            SimObjPhysics[] visibleSimObjs = null;
```

Comment:
As we're refactoring things anyway can we make it so that `VisibleSimObjs(ServerAction action)` does not take a `ServerAction` and instead explicitly declares its parameters?

Reply:
Done, i removed this method altogether and refactored all uses to `getInteractableSimObjectFromId`